### ISSUE TITLE
Fix "Quick Add Next" to match whole word only (quick variable selection)

### ISSUE THREAD

**Issue by dantonnoriega** [rstudio/rstudio#4955]
## TL;DR
Quick Add Next keyboard short cut should do a "whole word" match by default (like Sublime Text 3).

### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1522
    OS Version      :  macOS Mojave 10.14.5
    R Version       : R version 3.5.3 (2019-03-11)

### Steps to reproduce the problem

Quick add next of any variable that overlaps with a string that contains the same variable should reproduce the problem e.g. `post` and `posterior` will select `post` correctly but then also highlight `post` in `posterior`.

### Describe the problem in detail

I believe that, by default, "Quick Add Next" should be doing a whole word match. I think the purpose is to quickly select *variables* not just next.

I believe this used to work correctly but hasn't since 1.2.

In RStudio (Bad):
![quick-add-next-rstudio](https://user-images.githubusercontent.com/5673433/59528216-8341ae80-8e92-11e9-955d-7e253a408e4d.gif)

In Sublime Text 3 (Good):
![quick-add-next-st3](https://user-images.githubusercontent.com/5673433/59528231-8d63ad00-8e92-11e9-8c3a-bcd3528d543c.gif)

### Describe the behavior you expected
See the above gif for Sublime Text 3. But in words:

Assume the following code.

```
a <- 1:5
ab <- 1:100
a + ab + (a)^2
```

I would expect if I "Quick add next" the variable `a` (as a whole word), then I would select, in order:
- Line 1, first `a`
- Line 3, first `a` in the `a +`
- Line 3, third `a` within `(a)^2`

However, at the moment, Quick Add Next will go through and just select all the `a`s.

**Comment by dantonnoriega** [comment]
Related to #3344

**Comment by ronblum** [comment]
@dantonnoriega Thank you for raising this issue. We will review it as we work on development of future versions of RStudio.

Reproduced in RStudio Desktop 1.2.1547 with Sublime Text keybinding, and compared to Sublime Text 3.2.1 on MacOS 10.14.6.
